crates/sdk/src/sys/types/buffer.rs

```diff
@@ -41,7 +41,7 @@ impl<'a, T> Slice<'a, T> {
     // TODO: but this function is meant to be infallible. That is a concern, as
     // TODO: we want to eliminate all potential panics. This needs future
     // TODO: assessment.
-    #[allow(clippy::cast_possible_truncation)]
+    #[expect(clippy::cast_possible_truncation, reason = "TODO: See above")]
```

Comment:
this executes on the guest, which at the time of writing is wasm32, so `usize` will be a 32-bits

We use `u64` to accommodate the potential future effort of supporting `wasm64` as well

And finally, `Buffer`s will only be constructed on the guest, never on the host, so it should always be safe to convert whatever made up `self.len` (which was originally a `usize`) back to a `usize`.

Reply:
That would be great to note in the code 🙂